### ISSUE TITLE
No error for importing unknown module using `import "mod"` syntax

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#4561]
```ts
import "someUnknownFile";
```

expected to see an error for "can not find module "someUnknownFile"".

**Comment by weswigham** [comment]
If ES [loaders](https://whatwg.github.io/loader/) are definable at runtime, how can we suppose to know how an `import` is going to resolve? Are we just going to try to (simultaneously) support the `Browser` loader and the `Node` loaders? Or can we provide a facility for describing how a loader resolves a module name and finds the code associated with it?

**Comment by mhegazy** [comment]
the compiler emulates the resolution loigic for imports in general. so `import d from "myPackage"` will resolve to `node_modules\myPackage\index.d.ts` assuming you are using CommonJs, and will give you an error if you are using AMD/UMD/SystemJS.

the issue is our error reporting now depends on the typechecker needing the module symbol to check something against it. in the case of `import "mod"` the symbol is never needed, so the checking never happens.

**Comment by mhegazy** [comment]
Avoid the breaking change, and the current behavior is not impacting correctness.
